signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
> Well, the signature doesn’t just contain a tag which we happily apply to any manifest; it also contains a manifest digest. We will accept a signature which says sha256:HEX_1 and testimage:anythingwedontcare. The “anythingwedontcare” is the “any tag” in this terminology.

right, I know that, indeed I was asking: matchAnyTag really means "don't care about the tag in the signature docker reference but just match the digests if an user request an image@digest". See why I'm saying matchAnyTag is confusing? I would rather have something which actually says "matchAnyTagButMatchDigestPreciselyOfCourse" (joking but probably you get my point).
Right now people can think of that matchAnyTag as "docker pull busybox@digestforlatest" to be allowed when we have a signature for another tag, say latest1 which for us is obvious it has another digest and signature won't match.

> I am definitely open to other suggestions.

This is tricky then :( I guess we can still call it UntaggedReference and thoroughly document what it really means.